lib/pbench/server/filetree.py

```diff
@@ -903,3 +903,24 @@ def delete(self, dataset: str):
         tarball.controller.delete(dataset)
         del self.datasets[dataset]
         self._clean_empties(tarball.controller_name)
+
+    def find_inventory(self, dataset: str) -> Path:
+        """
+        Given the name of a dataset,search the INCOMING tree directory for
+        the dataset and return its relative path.
+
+        Args:
+            dataset: The name of a dataset that might exist somewhere in the
+                file tree
+
+        Raises:
+            DatasetNotFound: the INCOMING tree does not contain a dataset ,either
+            it is not unpacked or dataset is not found .
+
+        Returns:
+            A Path of the dataset directory.
+        """
+        for dirs in self.incoming_root.rglob("*"):
+            if dirs.name == dataset:
+                return dirs.absolute()
+        raise DatasetNotFound(dataset)
```

Comment:
Is this the appropriate exception to raise?  (We're raising this because a file system artifact is missing...if the _dataset_ were missing, we never would have gotten this far, right?)

Reply:
This, too, will be moot if you follow Dave's `find_dataset()` suggestion above.